test/unit_tests/models/test_util.py

```diff
@@ -20,3 +24,44 @@ def test_get_output_shape_2d_model():
         nn.Conv2d(1, 1, (3, 1)))
     out_shape = get_output_shape(model, in_chans=1, input_window_samples=5)
     assert out_shape == (1, 1, 3, 1)
+
+
+@pytest.mark.parametrize('dtype', [np.float16, np.float32, np.float64])
+@pytest.mark.parametrize('n_sequences,n_classes,n_windows,stride',
+                         [[10, 3, 2, 1], [3, 3, 2, 5], [3, 3, 1, 2]])
+def test_pad_shift_array(n_sequences, n_classes, n_windows, stride, dtype):
+    dense_y = np.random.RandomState(33).rand(
+        n_sequences, n_classes, n_windows).astype(dtype)
+
+    n_outputs = (n_sequences - 1) * stride + n_windows
+    shifted_y = np.concatenate([
+        np.concatenate((
+            np.zeros((1, n_classes, i * stride)), dense_y[[i]],
+            np.zeros((1, n_classes, n_outputs - n_windows - i * stride))),
+            axis=2)
+        for i in range(n_sequences)], axis=0)
+    shifted_y2 = _pad_shift_array(dense_y, stride=stride)
+
+    assert (shifted_y == shifted_y2).all()
```

Comment:
are you just testing zero arrays here? might be good to have somethign that tests actual values as well, like 1-2 more explicit tests with predefined values

Reply:
It is testing against actual values - the concatenation only adds zeros around the values in `dense_y`.